src/Microsoft.DotNet.Darc/DarcLib/ILocalGitRepo.cs

```diff
@@ -21,16 +23,43 @@ public interface ILocalGitRepo : IGitRepo
     ///     Checkout the repo to the specified state.
     /// </summary>
     /// <param name="repoDir">Path to a git repository</param>
-    /// <param name="commit">Tag, branch, or commit to checkout</param>
+    /// <param name="refToCheckout">Tag, branch, or commit to checkout</param>
     /// <param name="force">Force clean of the repo/submodules</param>
-    void Checkout(string repoDir, string commit, bool force = false);
+    void Checkout(string repoDir, string refToCheckout, bool force = false);
+
+    /// <summary>
+    ///     Commits files by calling git commit (not through Libgit2sharp)
+    /// </summary>
+    /// <param name="repoPath">Path of the local repository</param>
+    /// <param name="message">Commit message</param>
+    /// <param name="allowEmpty">Allow empty commits?</param>
+    /// <param name="identity">Identity object containing username and email. Defaults to DarcBot identity</param>
+    Task Commit(
+        string repoPath,
+        string message,
+        bool allowEmpty,
+        LibGit2Sharp.Identity? identity = null,
```

Comment:
We do use it across the project (we do for instance have the dotnet-bot identity in contants) and in this implementation take the email/name from it so I'd keep it instead of passing in the fields themselves.